src/app/models/types/onward-journey.ts

```diff
@@ -0,0 +1,126 @@
+import { OptimoBlock } from './optimo';
+
+export type OptimoRecommendation = {
+  headlines: {
+    seoHeadline?: string;
+    promoHeadline: {
+      blocks: OptimoBlock[];
+    };
+  };
+  locators: {
+    optimoUrn: string;
+    canonicalUrl: string;
+  };
+  summary: {
+    blocks: OptimoBlock[];
+  };
+  images: {
+    defaultPromoImage: {
+      blocks: OptimoBlock[];
+    };
+  };
+  timestamp: number;
+  suitableForSyndication?: boolean;
+  language?: string;
+  passport?: object;
+  id?: string;
+  type?: string;
+};
+
+export type CpsRecommendation = {
+  headlines: {
+    shortHeadline?: string;
+    headline: string;
+  };
+  locators: {
+    assetUri: string;
+    cpsUrn?: string;
+  };
+  summary: string;
+  timestamp: number;
+  indexImage: IndexImage;
+  cpsType?: string;
+  id?: string;
+  type?: string;
+};
+
+export type Recommendation = OptimoRecommendation | CpsRecommendation;
+
+export type TopStoriesOJ = {
+  headlines: Headlines;
+  locators: Locators;
+  summary: string;
+  timestamp: number;
+  language: string;
+  byline: Byline;
+  passport: Passport;
+  cpsType: string;
+  indexImage: IndexImage;
+  options: {
+    isBreakingNews: boolean;
+    isFactCheck: boolean;
+  };
+  prominence: string;
+  id: string;
+  type: string;
+};
+
+export type FeaturesAnalysisOJ = {
+  name: string;
+  summary: string;
+  indexImage: IndexImage;
+  uri: string;
+  aresUrl: string;
+  contentType: string;
+  assetTypeCode: string;
+  timestamp: number;
+  type: string;
+};
+
+export type Headlines = {
+  headline: string;
+};
+
+export type Locators = {
+  assetUri: string;
+  curieCpsUrn: string;
+  assetId: string;
+  cpsUrn: string;
+  curie: string;
+};
+
+export type Byline = {
+  name: string;
+  title: string;
+  persons?:
+    | {
+        name: string;
+        function: string;
+      }[]
+    | null;
```

Comment:
What is this TS magic 🪄 ?!?!

Reply:
Slight auto-formatting weirdness, but it basically says `campaigns` can be either:
```
{
   name: string;
   function: string;
}[]
```
or 
```
null
```